.circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```

Comment:
This is a nitpick and maybe something to address in a follow up PR, but I would much rather have the `TestInput.txt` file for this be in something like yaml or json so we don't have to go around trying to guess what each word in it is supposed to mean, as well as probably having this test be written in python to make it easier to process one of those formats.